data/trie/branchNode.go

```diff
@@ -419,9 +419,13 @@ func (bn *branchNode) delete(key []byte, db data.DBWriteCacher, marshalizer mars
 		return false, nil, err
 	}
 
-	dirty, newNode, err := bn.children[childPos].delete(key, db, marshalizer)
-	if !dirty || err != nil {
-		return false, nil, err
+	var newNode node
```

Comment:
Leave it as it was before, but add the following code before:
```	if bn.children[childPos] == nil {
		return false, bn, nil
	}
```

Reply:
done.